src/overlays/gamestates/ovl_file_choose/z_file_copy_erase.c

```diff
@@ -0,0 +1,1248 @@
+        this->buttonYOffsets[this->buttonIndex] = D_8081424C[this->buttonIndex][this->buttonIndex];
+        this->titleLabel = this->nextTitleLabel;
+        this->titleAlpha[0] = 255;
+        this->titleAlpha[1] = 0;
+        this->actionTimer = 4;
+        this->configMode++; // CM_SETUP_COPY_DEST_2
+    }
+}
+
+/**
+ * Show the file info of the file selected to copy from.
+ * Update function for `CM_SETUP_COPY_DEST_2`
+ */
+void FileSelect_SetupCopyDest2(GameState* thisx) {
+    FileSelectState* this = (FileSelectState*)thisx;
+
+    this->nameBoxAlpha[this->buttonIndex] -= 200 / 4;
+    this->fileInfoAlpha[this->buttonIndex] += 200 / 4;
+    this->actionTimer--;
+
+    if (this->actionTimer == 0) {
+        this->nameBoxAlpha[this->buttonIndex] = 0;
+        this->fileInfoAlpha[this->buttonIndex] = 200;
+        this->buttonIndex = FS_BTN_COPY_QUIT;
+        this->actionTimer = 4;
+        this->configMode = CM_SELECT_COPY_DEST;
+    }
+}
+
+/**
+ * Allow the player to select a slot to copy to or exit to the copy select screen.
+ * Update function for `CM_SELECT_COPY_DEST`
+ */
+void FileSelect_SelectCopyDest(GameState* thisx) {
+    FileSelectState* this = (FileSelectState*)thisx;
+    SramContext* sramCtx = &this->sramCtx;
+    Input* input = CONTROLLER1(&this->state);
+
+    if (((this->buttonIndex == FS_BTN_COPY_QUIT) && CHECK_BTN_ANY(input->press.button, BTN_A | BTN_START)) ||
+        CHECK_BTN_ALL(input->press.button, BTN_B)) {
+        this->buttonIndex = this->selectedFileIndex;
+        this->nextTitleLabel = FS_TITLE_COPY_FROM;
+        this->actionTimer = 4;
+        this->configMode = CM_EXIT_TO_COPY_SOURCE_1;
+        play_sound(NA_SE_SY_FSEL_CLOSE);
+    } else if (CHECK_BTN_ANY(input->press.button, BTN_A | BTN_START)) {
+        if (!gSaveContext.flashSaveAvailable) {
+            if (!NO_FLASH_SLOT_OCCUPIED(sramCtx, this->buttonIndex)) {
+                this->copyDestFileIndex = this->buttonIndex;
+                this->nextTitleLabel = FS_TITLE_COPY_CONFIRM;
+                this->actionTimer = 4;
+                this->configMode = CM_SETUP_COPY_CONFIRM_1;
+                play_sound(NA_SE_SY_FSEL_DECIDE_L);
+            } else {
+                play_sound(NA_SE_SY_FSEL_ERROR);
+            }
+        } else if (!SLOT_OCCUPIED(this, this->buttonIndex)) {
+            this->copyDestFileIndex = this->buttonIndex;
+            this->nextTitleLabel = FS_TITLE_COPY_CONFIRM;
+            this->actionTimer = 4;
+            this->configMode = CM_SETUP_COPY_CONFIRM_1;
+            play_sound(NA_SE_SY_FSEL_DECIDE_L);
+        } else {
+            play_sound(NA_SE_SY_FSEL_ERROR);
+        }
+    } else {
+        if (ABS_ALT(this->stickAdjY) >= 30) {
+            play_sound(NA_SE_SY_FSEL_CURSOR);
+            if (this->stickAdjY >= 30) {
+                this->buttonIndex--;
+                if (this->buttonIndex == FS_BTN_COPY_FILE_3) {
+                    this->buttonIndex = FS_BTN_COPY_FILE_2;
+                }
```

Comment:
It sure is yup

Reply:
It would be useful to annotate them, probably. Although I'd assume that JPN will have obvious changes around them, I'm sure someone will want to know prior to that.